### PR TITLE
Support int64 data types

### PR DESCRIPTION
**Pull request by otoolep**

### CODE REVIEW

**Comment by otoolep** [comment]
@benbjohnson

File: influxql/ast.go [hunk 1]
```diff
@@ -33,9 +35,9 @@ const (
 func InspectDataType(v interface{}) DataType {
 	switch v.(type) {
 	case float64:
-		return Number
-	case int:
-		return Number
+		return Float
+	case int64, int32, int:
```
Comment (otoolep) [inline influxql/ast.go]: Not sure if this is really necessary.

File: database.go [hunk 2]
```diff
@@ -780,17 +780,14 @@ func (f *FieldCodec) EncodeFields(values map[string]interface{}) ([]byte, error)
...
-			}
-
+		case influxql.Float:
+			value := v.(float64)
 			buf = make([]byte, 9)
 			binary.BigEndian.PutUint64(buf[1:9], math.Float64bits(value))
+		case influxql.Integer:
+			value := v.(int64)
+			buf = make([]byte, 9)
+			binary.BigEndian.PutUint64(buf[1:9], uint64(value))
```
Comment (otoolep) [inline database.go]: Seems easiest to just cast it to an unsigned value, and just encode it as that.

File: influxql/ast.go (hunk 1 above)
Comment (otoolep) [inline influxql/ast.go]: I might just need `int64`.

**Comment by dgnorton** [comment]
Can you add an integration test that writes & reads ints?

I tested this change on an ARM 32 bit system and it seems to work.  The output is interesting once we it gets above `math.MaxInt32` though.  See the values in the middle... https://gist.github.com/dgnorton/e734dc02cafc80241ae6.  It's not a CLI formatting thing.  That's also the way the values look in the JSON if I use curl.  Haven't tried this test on 64 bit yet.

**Comment by otoolep** [comment]
Thanks @dgnorton -- I'll add some more tests. The formatting issue you pointed out, yeah, I've seen that before. I am assuming it's due to the use of `float64`. It's nothing to do with this change.

File: server_test.go [hunk 3]
```diff
@@ -976,6 +976,39 @@ func TestServer_StartRetentionPolicyEnforcement_ErrZeroInterval(t *testing.T) {
 	}
 }
 
+// Ensure the server respects limit and offset in show series queries
+func TestServer_WriteAllDataTypes(t *testing.T) {
```
Comment (otoolep) [inline server_test.go]: Must add test to `server.go` since it's the only way to control the types.

**Comment by otoolep** [comment]
Updated with unit test. +1 @dgnorton ?

**Comment by dgnorton** [comment]
Why can it only be tested through server.go?  Users read & write int64 values through the usual API.

**Comment by otoolep** [comment]
If by "usual API" you mean they write through the JSON API, Go's JSON unmarshalling casts all numbers to float64.

http://golang.org/pkg/encoding/json/#Unmarshal

There is no way to get integer values into the system through the JSON API. It'll need an API change, or perhaps some code on our side.

**Comment by dgnorton** [comment]
I think we used to do it using `UseNumber()` maybe?  Not sure, have to go back and look at v0.8 code.

**Comment by otoolep** [comment]
You could be right @dgnorton -- but I think that is orthogonal to this change. Regardless of *how* integers get into the system, we need to have support for encoding them. This is what this change does.

**Comment by dgnorton** [comment]
lgtm  :shipit:

**Pull request merged**